- Fix build with OpenEXR>=3
- Switch to C++20
- Update Cmake and deps, Fix Freetype and FontConfig Linkage,
  from upstream
- disable GSL support for now. devel/gsl needs an update

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -105,11 +105,12 @@ endif()
 message(STATUS "Release build: ${RELEASE_BUILD}")
 
 # use CPP-11
-if (CMAKE_VERSION VERSION_LESS "3.1")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-else ()
-    set (CMAKE_CXX_STANDARD 11)
-endif ()
+#if (CMAKE_VERSION VERSION_LESS "3.1")
+#    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+#else ()
+#    set (CMAKE_CXX_STANDARD 11)
+#endif ()
+set (CMAKE_CXX_STANDARD 20)
 
 ############
 #############
@@ -391,21 +392,27 @@ endif ()
 ## Check for OpenEXR
 ##
 
-macro_optional_find_package(OpenEXR)
-macro_bool_to_01(OPENEXR_FOUND HAVE_OPENEXR)
+macro_optional_find_package(Imath 3.0 CONFIG QUIET)
+if(TARGET Imath::Imath)
+    set(OPENEXR_LIBRARIES Imath::Imath)
+    set(OpenEXR_FOUND TRUE)
+else()
+    macro_optional_find_package(OpenEXR)
+endif()
+macro_bool_to_01(OpenEXR_FOUND HAVE_OPENEXR)
 
 ##
 ## Test for GNU Scientific Library
 ##
-macro_optional_find_package(GSL 1.7)
-set_package_properties(GSL_FOUND PROPERTIES
-    DESCRIPTION "GNU Scientific Library"
-    URL "https://www.gnu.org/software/gsl"
-    PURPOSE "Required by Sheets' solver plugin"
-    TYPE OPTIONAL
-    )
-macro_bool_to_01(GSL_FOUND HAVE_GSL)
-configure_file(config-gsl.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-gsl.h )
+#macro_optional_find_package(GSL 1.7)
+#set_package_properties(GSL_FOUND PROPERTIES
+#    DESCRIPTION "GNU Scientific Library"
+#    URL "https://www.gnu.org/software/gsl"
+#    PURPOSE "Required by Sheets' solver plugin"
+#    TYPE OPTIONAL
+#    )
+#macro_bool_to_01(GSL_FOUND HAVE_GSL)
+#configure_file(config-gsl.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-gsl.h )
 
 ##
 ## Test for Phonon4Qt5
@@ -641,8 +648,8 @@ if(NOT WIN32 AND NOT APPLE)
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
@@ -912,6 +919,8 @@ set(KOPAGEAPP_INCLUDES ${TEXTLAYOUT_INCLUDES}
     ${CMAKE_SOURCE_DIR}/libs/pageapp
     ${CMAKE_SOURCE_DIR}/libs/pageapp/commands
     ${CMAKE_BINARY_DIR}/libs/pageapp )
+
+
 
 #############################################
 ####           filter libraries          ####
